<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Run mash analysis on dreamlet results"><title>Run mash analysis on dreamlet results — run_mash • dreamlet</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Run mash analysis on dreamlet results — run_mash"><meta property="og:description" content="Run mash analysis on dreamlet results"><meta property="og:image" content="http://gabrielhoffman.github.io/dreamlet/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">dreamlet</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.51</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/dreamlet.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/h5ad_on_disk.html">Handling large H5AD datasets</a>
    <a class="dropdown-item" href="../articles/mashr.html">mashr analysis after dreamlet</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/GabrielHoffman/dreamlet/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Run mash analysis on dreamlet results</h1>
      <small class="dont-index">Source: <a href="https://github.com/GabrielHoffman/dreamlet/blob/HEAD/R/mash.R" class="external-link"><code>R/mash.R</code></a></small>
      <div class="d-none name"><code>run_mash.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Run mash analysis on dreamlet results</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_mash</span><span class="op">(</span><span class="va">fit</span>, <span class="va">coefList</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>fit</dt>
<dd><p>result from <code><a href="dreamlet.html">dreamlet()</a></code></p></dd>


<dt>coefList</dt>
<dd><p>coefficient to be analyzed</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>a list storing the <code>mashr</code> model as <code>model</code> and the original coefficients as <code>logFC.original</code></p>


    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Apply <a href="https://cran.r-project.org/web/packages/mashr/index.html" class="external-link">mashr</a> analysis (Urbut et al. 2019)
 on the joint set of coefficients for each gene and cell type.  <code>mashr</code> is a Bayesian statistical method that borrows strength across tests (i.e. genes and cell types) by learning the distribution of non-zero effects based the obesrved logFC and standard errors.  The method then estimates the posterior distributions of each coefficient based on the observed value and the genome-wide emprical distribution.</p>
<p><code>mashr</code> has been previously applied to differential expression in <a href="https://www.gtexportal.org" class="external-link">GTEx</a> data using multiple tissues from the same set of donors (Oliva et al. 2020)
.</p>
<p>In single cell data, a given gene is often not sufficiently expressed in all cell types.  So it is not evaluated in a subsets of cell types, and its coefficient value is <code>NA</code>. Since mashr assumes coefficients and standard errors for every gene and cell type pair, entries with these missing values are set to have <code>coef = 0</code>, and <code>se = 1e6</code>.  The output of mashr is then modified to set the corresponding values to <code>NA</code>, to avoid nonsensical results downstream.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Oliva M, Munoz-Aguirre M, Kim-Hellmuth S, Wucher V, Gewirtz AD, Cotter DJ, Parsana P, Kasela S, Balliu B, Vinuela A, others (2020).
“The impact of sex on gene expression across human tissues.”
<em>Science</em>, <b>369</b>(6509), eaba3066.
<a href="https://doi.org/10.1126/science.aba3066" class="external-link">https://doi.org/10.1126/science.aba3066</a>.<br><br> Urbut SM, Wang G, Carbonetto P, Stephens M (2019).
“Flexible statistical methods for estimating and testing effects in genomic studies with multiple conditions.”
<em>Nature genetics</em>, <b>51</b>(1), 187--195.
<a href="https://doi.org/10.1038/s41588-018-0268-8" class="external-link">https://doi.org/10.1038/s41588-018-0268-8</a>.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/mashr/man/mash_estimate_corr_em.html" class="external-link">mashr::mash_estimate_corr_em()</a></code>, <code><a href="https://rdrr.io/pkg/mashr/man/cov_canonical.html" class="external-link">mashr::cov_canonical</a></code>, <code><a href="https://rdrr.io/pkg/mashr/man/mash_set_data.html" class="external-link">mashr::mash_set_data</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/HelenaLC/muscat" class="external-link">muscat</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/stephenslab/mashr" class="external-link">mashr</a></span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: ashr</span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">SingleCellExperiment</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">example_sce</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># create pseudobulk for each sample and cell cluster</span></span></span>
<span class="r-in"><span><span class="va">pb</span> <span class="op">&lt;-</span> <span class="fu"><a href="aggregateToPseudoBulk.html">aggregateToPseudoBulk</a></span><span class="op">(</span><span class="va">example_sce</span>, </span></span>
<span class="r-in"><span>  assay <span class="op">=</span> <span class="st">"counts"</span>,    </span></span>
<span class="r-in"><span>  cluster_id <span class="op">=</span> <span class="st">'cluster_id'</span>, </span></span>
<span class="r-in"><span>  sample_id <span class="op">=</span> <span class="st">'sample_id'</span>,</span></span>
<span class="r-in"><span>  verbose<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># voom-style normalization</span></span></span>
<span class="r-in"><span><span class="va">res.proc</span> <span class="op">=</span> <span class="fu"><a href="processAssays.html">processAssays</a></span><span class="op">(</span> <span class="va">pb</span>, <span class="op">~</span> <span class="va">group_id</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   B cells...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.023 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   CD14+ Monocytes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.022 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   CD4 T cells...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.023 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   CD8 T cells...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.02 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   FCGR3A+ Monocytes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.023 secs</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Differential expression analysis within each assay,</span></span></span>
<span class="r-in"><span><span class="co"># evaluated on the voom normalized data </span></span></span>
<span class="r-in"><span><span class="va">res.dl</span> <span class="op">=</span> <span class="fu"><a href="dreamlet.html">dreamlet</a></span><span class="op">(</span> <span class="va">res.proc</span>, <span class="op">~</span> <span class="va">group_id</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   B cells...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.044 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   CD14+ Monocytes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.067 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   CD4 T cells...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.048 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   CD8 T cells...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.03 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>   FCGR3A+ Monocytes...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 0.062 secs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># run MASH model</span></span></span>
<span class="r-in"><span><span class="co"># This can take 10s of minutes on real data</span></span></span>
<span class="r-in"><span><span class="co"># This small datasets should take ~30s</span></span></span>
<span class="r-in"><span><span class="va">res_mash</span> <span class="op">=</span> <span class="fu">run_mash</span><span class="op">(</span> <span class="va">res.dl</span>, <span class="st">'group_idstim'</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in data.frame(assay = k, tab):</span> arguments imply differing number of rows: 1, 0</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># extract statistics from mashr model</span></span></span>
<span class="r-in"><span><span class="co"># NA values indicate genes not sufficiently expressed</span></span></span>
<span class="r-in"><span><span class="co"># in a given cell type</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># original logFC</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">$</span><span class="va">logFC.original</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'head': object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># posterior mean for logFC</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ashr/man/get_lfdr.html" class="external-link">get_pm</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">$</span><span class="va">model</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'head': object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># how many gene-by-celltype tests are significant</span></span></span>
<span class="r-in"><span><span class="co"># i.e.  if a gene is significant in 2 celltypes, it is counted twice</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ashr/man/get_lfdr.html" class="external-link">get_lfsr</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">$</span><span class="va">model</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, useNA<span class="op">=</span><span class="st">"ifany"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in get_lfsr(res_mash$model):</span> object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># how many genes are significant in at least one cell type</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ashr/man/get_lfdr.html" class="external-link">get_lfsr</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">$</span><span class="va">model</span><span class="op">)</span>, <span class="fl">1</span>, <span class="va">min</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in get_lfsr(res_mash$model):</span> object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># how many genes are significant in each cell type</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/ashr/man/get_lfdr.html" class="external-link">get_lfsr</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">$</span><span class="va">model</span><span class="op">)</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">x</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, na.rm<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in get_lfsr(res_mash$model):</span> object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># examine top set of genes</span></span></span>
<span class="r-in"><span><span class="co"># which genes are significant in at least 1 cell type</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/mashr/man/get_significant_results.html" class="external-link">get_significant_results</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">$</span><span class="va">model</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'sort': object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Lets examine CXCL3</span></span></span>
<span class="r-in"><span><span class="co"># There is a lot of variation in the raw logFC</span></span></span>
<span class="r-in"><span><span class="va">res_mash</span><span class="op">$</span><span class="va">logFC.original</span><span class="op">[</span><span class="st">"ISG20"</span>,<span class="op">]</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in eval(expr, envir, enclos):</span> object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># posterior mean after borrowing across cell type and genes</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/ashr/man/get_lfdr.html" class="external-link">get_pm</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">$</span><span class="va">model</span><span class="op">)</span><span class="op">[</span><span class="st">"ISG20"</span>,<span class="op">]</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in get_pm(res_mash$model):</span> object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># gene set analysis using mashr results</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://DiseaseNeuroGenomics.github.io/zenith" class="external-link">zenith</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">go.gs</span> <span class="op">=</span> <span class="fu"><a href="http://gabrielhoffman.github.io/zenith/reference/get_GeneOntology.html" class="external-link">get_GeneOntology</a></span><span class="op">(</span><span class="st">"CC"</span>, to<span class="op">=</span><span class="st">"SYMBOL"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span><span class="co"># valid values for statistic: </span></span></span>
<span class="r-in"><span><span class="co"># "tstatistic", "abs(tstatistic)", "logFC", "abs(logFC)"</span></span></span>
<span class="r-in"><span><span class="va">df_gs</span> <span class="op">=</span> <span class="fu"><a href="http://gabrielhoffman.github.io/zenith/reference/zenith_gsa-methods.html" class="external-link">zenith_gsa</a></span><span class="op">(</span><span class="va">res_mash</span>, <span class="va">go.gs</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'fit' in selecting a method for function 'zenith_gsa': object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Heatmap of results</span></span></span>
<span class="r-in"><span><span class="fu"><a href="http://gabrielhoffman.github.io/zenith/reference/plotZenithResults.html" class="external-link">plotZenithResults</a></span><span class="op">(</span><span class="va">df_gs</span>, <span class="fl">2</span>, <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in is.data.frame(x):</span> object 'df_gs' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># forest plot based on mashr results</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotForest-methods.html">plotForest</a></span><span class="op">(</span><span class="va">res_mash</span>, <span class="st">"ISG20"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'plotForest': object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># volcano plot based on mashr results</span></span></span>
<span class="r-in"><span><span class="co"># yaxis uses local false sign rate (lfsr)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="plotVolcano-methods.html">plotVolcano</a></span><span class="op">(</span><span class="va">res_mash</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in h(simpleError(msg, call)):</span> error in evaluating the argument 'x' in selecting a method for function 'plotVolcano': object 'res_mash' not found</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="http://gabrielhoffman.github.io" class="external-link">Gabriel Hoffman</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

